lib/spack/spack/concretize.py

```diff
@@ -465,6 +472,54 @@ def _compiler_concretization_failure(compiler_spec, arch):
         raise UnavailableCompilerVersionError(compiler_spec, arch)
 
 
+def concretize_specs_together(*abstract_specs):
+    """Given a number of specs as input, tries to concretize them together.
+
+    Args:
+        *abstract_specs: abstract specs to be concretized
+
+    Returns:
+        List of concretized specs
+    """
+    # Create a temporary repository with an umbrella package
+    # that depends on each spec
+    tmpdir = tempfile.mkdtemp()
```

Comment:
Could we create a separate helper function for the repo setup? The rest of the function only needs the actual repo object. I think keeping the spec-management logic separate from the repo-creation-from-template logic will make the code a lot easier to read.

Then we could just call `with spack.repo.additional_repository(_make_concretization_repo(split_specs)):`

Reply:
I created a nested helper function, as it won't be used elsewhere even within this module. Let me know if that's fine with you.